pkg/generator/gen_test.go

```diff
@@ -0,0 +1,173 @@
+/*
+Copyright The KCP Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package generator
+
+import (
+	"testing"
+
+	"k8s.io/code-generator/cmd/client-gen/types"
+
+	"github.com/kcp-dev/client-gen/pkg/flag"
+	. "github.com/onsi/ginkgo"
```

Comment:
Would prefer not using ginkgo, but I don't necessarily want to block getting this in. @sttts @stevekuznetsov WDYT?

Reply:
Sure, can I rewrite them in built-in go test framework in a follow-up.